bootstrap/templates/ansible/playbooks/cluster-rollout-update.yaml.j2

```diff
@@ -25,46 +25,45 @@
+        - name: "Cordon {% raw %}{{ inventory_hostname }}{% endraw %}"
+          kubernetes.core.k8s_drain:
+            state: cordon
+            name: "{% raw %}{{ inventory_hostname }}{% endraw %}"
           delegate_to: "{% raw %}{{ groups['master'][0] }}{% endraw %}"
-          changed_when: false
 
-        - name: Wait for node to cordon
-          ansible.builtin.command: "kubectl get node {% raw %}{{ inventory_hostname }}{% endraw %} -o json"
-          register: wait_for_cordon
-          retries: 10
-          delay: 10
+        - name: "Drain {% raw %}{{ inventory_hostname }}{% endraw %}"
+          kubernetes.core.k8s_drain:
+            state: drain
+            name: {% raw %}{{ inventory_hostname }}{% endraw %}
+            delete_options:
+              delete_emptydir_data: true
+              ignore_daemonsets: true
+              terminate_grace_period: 60
+              wait_timeout: 120
+          # Supported feature in upcoming kubernetes.core 2.5.0
+          #  pod_selectors:
+          #    - app!=rook-ceph-osd
           delegate_to: "{% raw %}{{ groups['master'][0] }}{% endraw %}"
-          changed_when: false
-          until: (wait_for_cordon['stdout'] | from_json).spec.unschedulable
-
-        - name: Cordon & drain node
-          ansible.builtin.command: "kubectl drain --ignore-daemonsets --delete-emptydir-data --force {% raw %}{{ inventory_hostname }}{% endraw %}"
-          delegate_to: "{% raw %}{{ groups['master'][0] }}{% endraw %}"
-          changed_when: false
 
         - name: Update all packages
           ansible.builtin.apt:
             upgrade: dist
             update_cache: true
 
+        - name: Check if a Reboot is Required
+          ansible.builtin.stat:
+            path: /var/run/reboot-required
+          register: reboot_required
+
         - name: Reboot
           ansible.builtin.reboot:
             msg: Rebooting node
             post_reboot_delay: 60
             reboot_timeout: 3600
+          when: reboot_required.stat.exists
 
-        - name: Uncordon node
-          ansible.builtin.command: "kubectl uncordon {% raw %}{{ inventory_hostname }}{% endraw %}"
-          delegate_to: "{% raw %}{{ groups['master'][0] }}{% endraw %}"
-          changed_when: false
-
-        - name: Wait for node to uncordon
-          ansible.builtin.command: "kubectl get node {% raw %}{{ inventory_hostname }}{% endraw %} -o json"
-          register: wait_for_uncordon
-          retries: 10
-          delay: 10
-          delegate_to: "{% raw %}{{ groups['master'][0] }}{% endraw %}"
-          changed_when: false
-          until: not (kubectl_get_node['stdout'] | from_json).spec.unschedulable is defined
+        - name: "Uncordon {% raw %}{{ inventory_hostname }}{% endraw %}"
+          kubernetes.core.k8s_drain:
+            state: uncordon
+            name: "{% raw %}{{ inventory_hostname }}{% endraw %}"
+          delegate_to: localhost
```

Comment:
Should this be delegated to `{{ groups['master'][0] }}` like the others?

Reply:
Yep, committed fix for this. I have been using my dev box (localhost) to run my role on cron, didn't even think to use one of the control plane nodes.